.github/workflows/build-publish-documentation.yaml

```diff
@@ -72,9 +69,11 @@ jobs:
           # generate HTML from markdown and put into dev version
           poetry run mike deploy dev
           
+          # ensure the main branch is available locally
+          git fetch origin main
           # switch to gh-pages branch, copy over the latest 404.html from main
           git checkout --force gh-pages
-          git checkout --force main -- 404.html
+          git checkout --force origin/main -- 404.html
```

Comment:
I use mike for multi-versioning in [nfdi4cat/pid4cat-model](https://github.com/nfdi4cat/pid4cat-model) and the checkout acrobatics and the copying of the 404.html here are strange. I added a [proper 404.html](https://github.com/nfdi4cat/pid4cat-model/blob/gh-pages/404.html) to the gh-pages branch and never hat to touch it again.

However, if it works here then maybe leave it as it is.

Reply:
I'm responsible for the checkout acrobatics here! 😂 

I'm afraid I don't quite remember all the details on why it is the way it is. There is a brief comment in https://github.com/linkml/linkml-model/pull/168 about it. But ultimately I think I just didn't like the idea of a file _only_ living in the `gh-pages` branch where it could be potentially forgotten about.